### ISSUE TITLE
Add linter to enforce style to code

### ISSUE THREAD

**Issue by fredericomba** [Yuri-SVB/Great_Wall#74]
There must be a linter for enforcing the absence of some patterns in the code  Notably, this one was rejected by Yuri:
```Python
# wrong
if variable: # implicit use of falsy values
  pass

# correct
if variable != 0:
  pass
```

**Comment by MuhammadMuradG** [comment]
Thanks for your reporting! Actually, we intend to address this problem and other similar ones in the process of testing. Specifically, we intend to use:
* [Black](https://black.readthedocs.io/en/stable/index.html) ensures uniform code formatting
* [docformatter](https://docformatter.readthedocs.io/en/latest/) ensures uniform formatting for docstrings and comments
* [pyupgrade](https://github.com/asottile/pyupgrade) ensures code is using the latest best practices for Python
* [isort](https://pycqa.github.io/isort/) ensures uniform import statements
* [flake8](https://flake8.pycqa.org/en/latest/) checks for common coding and syntax issues

**Comment by MuhammadMuradG** [comment]
Of course, these tools are initial suggestions and we are open to any discussion and another suggestions.
